### ISSUE TITLE
Modelling

### ISSUE THREAD

**Issue by eeshawn11** [eeshawn11/DSI33-Project4#3]
1. Pre-processing
2. Modelling
3. Hyperparameter tuning

**Comment by eeshawn11** [comment]
Attempted predictions with various models, scoring based on `ROC AUC` metric in sklearn.
Created an ensemble with the 3 best performing models

model_list | mean_train_score | mean_test_score | score_delta | best_params | runtime
-- | -- | -- | -- | -- | --
VotingClassifier | 0.8374 | 0.7846 | 0.0528 | {'clf__GradientBoost__learning_rate': 0.1, 'cl... | 455.15
GradientBoostingClassifier | 0.8294 | 0.7777 | 0.0517 | {'clf__learning_rate': 0.1, 'clf__max_depth': ... | 176.89
LogisticRegression | 0.8236 | 0.7684 | 0.0552 | {'clf__C': 10, 'clf__solver': 'liblinear'} | 27.55
RidgeClassifier | 0.8171 | 0.7589 | 0.0582 | {'clf__alpha': 0.9} | 33.16
AdaBoostClassifier | 0.8169 | 0.7702 | 0.0468 | {'clf__learning_rate': 1.0} | 18.00
RandomForestClassifier | 0.8135 | 0.7825 | 0.0310 | {'clf__ccp_alpha': 0.01, 'clf__max_features': ... | 70.52
Perceptron | 0.8052 | 0.7489 | 0.0563 | {'clf__class_weight': None, 'clf__penalty': None} | 21.88
KNeighborsClassifier | 0.7930 | 0.7330 | 0.0600 | {'clf__n_neighbors': 15, 'clf__weights': 'unif... | 34.01

Kaggle Score
- Score: 0.54786
- Private score: 0.53161
